magma/bits.py

```diff
@@ -40,6 +40,15 @@ class BitsMeta(AbstractBitVectorMeta, ArrayMeta):
     def __new__(mcs, name, bases, namespace, info=(None, None, None), **kwargs):
         return ArrayMeta.__new__(mcs, name, bases, namespace, info, **kwargs)
 
+    def __call__(cls, *args, **kwargs):
+        result = super().__call__(*args, **kwargs)
+        if all(x.is_output() for x in result.ts) and not cls.is_output():
+            if cls.is_input() or cls.is_inout():
```

Comment:
how is this possible?

Reply:
This would happen if the user said `m.In(Bits[5])(value)` where value contained outputs.